### DISCUSSION TITLE
How can I run an application with a flag that contains a reference?

### DISCUSSION THREAD

**Discussion by ma3103yd-s** [hecrj/iced#854] [Q&A]
Basically I am trying to send a struct containing a reference to a dynamic matrix as a flag when running the application. I am getting an error because it says my reference needs to be static. Below is my code. 
```
    let x = Linspace::linspace_f32(0.0, 5.0, 10);
    let y = Linspace::linspace_f32(0.0, 5.0, 10);
    let (X, Y) = meshgrid(&x,&y);
    let Z = X.component_mul(&X)+Y.component_mul(&Y);
    
    let s = Surface3D::new(&X,&Y,&Z);
    let plot = Plot3D::plot(s);

    Window3D::run(Settings{
        window: window::Settings::default(),
        flags: plot,
        default_font: None,
        default_text_size: 20,
        antialiasing: true,
    });
```
Basically Surface3D is a struct containing references to my matrices X,Y and Z and Plot3D is another struct containing a Surface3D. I am then trying to send my plot as a flag to my Window3D which implements application.

**Comment by H36615** [comment]
I'm still learning rust but I was able to supply static reference of struct to flags. From the source code of `run`:
```
pub trait Application: Sized {
    ...
    fn run(settings: Settings<Self::Flags>) -> crate::Result
    where
        Self: 'static,
    {
```
So if i'm not mistaken, the Window3D needs to be static meaning that its field that you want to assign a reference needs also to be static type. From my code:
```
struct Tester {
    myStruct: &'static MyStruct,
}

impl Application for Tester {
    ...
    fn new(flags: Flags) -> (Self, Command<Message>) {
        (
            Self {
                myStruct: flags.myStruct,
            },
            Command::none(),
        )
    }
```
Then to start the UI: `Tester::run(staticReferenceToMyStruct);`

Please correct me if said something wrong.

**Reply by ma3103yd-s** [reply]
Thanks for the help! I appreciate it.
